handlers/oauthhandler/oauthhandler.go

```diff
@@ -0,0 +1,181 @@
+package oauthhandler
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/vocdoni/blind-csp/model"
+	"github.com/vocdoni/blind-csp/types"
+	"go.vocdoni.io/dvote/log"
+)
+
+// OauthHandler is a handler that requires a verifiable oAuth token to be resolved.
+type OauthHandler struct{}
+
+// Init does nothing
+func (oh *OauthHandler) Init(opts ...string) error {
+	return nil
+}
+
+// GetName returns the name of the handler
+func (oh *OauthHandler) Name() string {
+	return "oAuth"
+}
+
+// Info returns the handler options and required auth steps.
+func (oh *OauthHandler) Info() *types.Message {
+	return &types.Message{
+		Title:    "oAuth handler",
+		AuthType: "auth",
+		SignType: types.AllSignatures,
+		AuthSteps: []*types.AuthField{
+			{Title: "GetAuthUrl", Type: "text"},
+			{Title: "VerifyElection", Type: "text"},
+		},
+	}
+}
+
+// Indexer takes a unique user identifier and returns the list of processIDs where
+// the user is elegible for participation. This is a helper function that might not
+// be implemented (depends on the handler use case).
+func (oh *OauthHandler) Indexer(userID types.HexBytes) []types.Election {
```

Comment:
Consider implementing this handler.

The use case is a participant (voter) that wants to know on which ongoing CSP voting elections can he participate.  This would help the UI, since a user identified by an ID (the handler chooses how this ID is created) just by providing it, can obtain the list of elections where he can cast a vote.

Ideally, this ID would be something derived from the Oauth data and the handler should be able to find the existing elections for it.

Please don't hesitate to discard this if you think its not appropriate.

Reply:
This is an interesting use case, but I think is out of the scope for this PR.
Happy to talk about it as I think it has some details to be discussed